tests/unit/pubsub_v1/publisher/test_publisher_client.py

```diff
@@ -129,6 +135,222 @@ def test_init_w_custom_transport(creds):
+    client._get_or_create_sequencer = mock.Mock(return_value=sequencer)
+
+    TOPIC = "projects/projectID/topics/topicID"
+    with pytest.raises(RuntimeError):
+        client.publish(TOPIC, b"message")
+
+    spans = span_exporter.get_finished_spans()
+
+    if enable_open_telemetry:
+        # Span 1: Publisher Flow Control span
+        # Span 2: Publisher Batching span
+        # Span 3: Create Publish span
+        assert len(spans) == 3
+
+        flow_control_span, batching_span, create_span = spans
+
+        # Verify batching span contents.
+        assert batching_span.name == "publisher batching"
+        assert batching_span.kind == trace.SpanKind.INTERNAL
+        assert batching_span.parent.span_id == create_span.get_span_context().span_id
+
+        # Verify exception recorded by the publisher batching span.
+        assert batching_span.status.status_code == trace.StatusCode.ERROR
+        assert len(batching_span.events) == 1
+        assert batching_span.events[0].name == "exception"
+
+        # Verify exception recorded by the publisher create span.
+        assert create_span.status.status_code == trace.StatusCode.ERROR
+        assert len(create_span.events) == 2
+        assert create_span.events[0].name == "publish start"
+        assert create_span.events[1].name == "exception"
+
+        # Verify the finished flow control span.
+        assert flow_control_span.name == "publisher flow control"
+        assert len(flow_control_span.events) == 0
+    else:
+        assert len(spans) == 0
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 8),
+    reason="Open Telemetry not supported below Python version 3.8",
+)
+def test_opentelemetry_flow_control_exception(creds, span_exporter):
+    publisher_options = types.PublisherOptions(
+        flow_control=types.PublishFlowControl(
+            message_limit=10,
+            byte_limit=150,
+            limit_exceeded_behavior=types.LimitExceededBehavior.ERROR,
+        ),
+        enable_open_telemetry_tracing=True,
+    )
+    client = publisher.Client(credentials=creds, publisher_options=publisher_options)
+
+    mock_batch = mock.Mock(spec=client._batch_class)
+    topic = "projects/projectID/topics/topicID"
+    client._set_batch(topic, mock_batch)
+
+    future1 = client.publish(topic, b"a" * 60)
+    future2 = client.publish(topic, b"b" * 100)
+
+    future1.result()  # no error, still within flow control limits
+    with pytest.raises(exceptions.FlowControlLimitError):
+        future2.result()
+
+    spans = span_exporter.get_finished_spans()
+    # Span 1 = Publisher Flow Control Span of first publish
+    # Span 2 = Publisher Batching Span of first publish
+    # Span 3 = Publisher Flow Control Span of second publish(raises FlowControlLimitError)
+    # Span 4 = Publish Create Span of second publish(raises FlowControlLimitError)
+    assert len(spans) == 4
+
+    failed_flow_control_span = spans[2]
+    finished_publish_create_span = spans[3]
```

Comment:
Is it possible these spans might be created in a different order?

Reply:
No, because I create / end them in a particular order.